### ISSUE TITLE
9.1.0-rc1 Fails to build

### ISSUE THREAD

**Issue by sebastic** [OSGeo/proj.4#3295]
#### Example of problem
PROJ 9.1.0-rc1 fails to build:
```
[  1%] Linking C executable ../../bin/geodsigntest
cd "/build/proj-9.1.0~rc1/obj-x86_64-linux-gnu/src/tests" && /usr/bin/cmake -E cmake_link_script CMakeFiles/geodsigntest.dir/link.txt --verbose=1
/usr/bin/cc -g -O2 -ffile-prefix-map=/build/proj-9.1.0~rc1=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -Wl,-z,relro -Wl,-z,now CMakeFiles/geodsigntest.dir/geodsigntest.c.o -o ../../bin/geodsigntest   -L"/build/proj-9.1.0~rc1/obj-x86_64-linux-gnu/lib" 
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `areareduceB':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:2000: undefined reference to `remainder'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `AngDiff':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:152: undefined reference to `remainder'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:152: undefined reference to `remainder'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:155: undefined reference to `remainder'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `sincosdx':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:179: undefined reference to `remquo'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:181: undefined reference to `sincos'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `atan2dx':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:226: undefined reference to `atan2'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:226: undefined reference to `atan2'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:226: undefined reference to `atan2'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:226: undefined reference to `atan2'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `AngNormalize':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:142: undefined reference to `remainder'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:142: undefined reference to `remainder'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:142: undefined reference to `remainder'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `norm2':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:135: undefined reference to `hypot'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `geod_lineinit_int':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:352: undefined reference to `fmax'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:359: undefined reference to `hypot'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `norm2':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:135: undefined reference to `hypot'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `SinCosSeries':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:1101: undefined reference to `sincos'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `geod_lineinit_int':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:375: undefined reference to `sqrt'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:353: undefined reference to `sqrt'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `sincosde':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:200: undefined reference to `remquo'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:202: undefined reference to `sincos'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `norm2':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:135: undefined reference to `hypot'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `geod_geninverse_int':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:760: undefined reference to `fmax'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `norm2':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:135: undefined reference to `hypot'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `geod_geninverse_int':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:764: undefined reference to `fmax'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:837: undefined reference to `sincos'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:801: undefined reference to `fmax'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:801: undefined reference to `atan2'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:955: undefined reference to `hypot'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:977: undefined reference to `sincos'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:1005: undefined reference to `atan2'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `InverseStart':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:1248: undefined reference to `sincos'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:1259: undefined reference to `hypot'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `norm2':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:135: undefined reference to `hypot'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `InverseStart':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:1269: undefined reference to `atan2'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `norm2':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:135: undefined reference to `hypot'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:135: undefined reference to `hypot'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `sq':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:99: undefined reference to `sincos'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `Lambda12':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:1407: undefined reference to `hypot'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `norm2':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:135: undefined reference to `hypot'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:135: undefined reference to `hypot'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `Lambda12':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:1439: undefined reference to `fmax'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:1439: undefined reference to `atan2'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:1443: undefined reference to `fmax'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:1446: undefined reference to `atan2'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `geod_geninverse_int':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:899: undefined reference to `sincos'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `norm2':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:135: undefined reference to `hypot'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `geod_geninverse_int':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:711: undefined reference to `sincos'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `norm2':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:135: undefined reference to `hypot'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `geod_geninverse_int':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:990: undefined reference to `atan2'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `norm2':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:135: undefined reference to `hypot'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `InverseStart':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:1278: undefined reference to `atan2'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:1314: undefined reference to `fmax'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `Astroid':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:1192: undefined reference to `cbrt'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `InverseStart':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:1354: undefined reference to `sincos'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:1295: undefined reference to `atan2'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `Astroid':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:1197: undefined reference to `atan2'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:1200: undefined reference to `cos'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `InverseStart':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:1312: undefined reference to `fmin'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `Lambda12':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:1449: undefined reference to `sqrt'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `geod_geninverse_int':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:782: undefined reference to `sqrt'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:783: undefined reference to `sqrt'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `Lambda12':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:1426: undefined reference to `sqrt'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `Astroid':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:1202: undefined reference to `sqrt'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o:./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:1208: more undefined references to `sqrt' follow
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `geod_init':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:309: undefined reference to `atanh'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:321: undefined reference to `fmax'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:321: undefined reference to `fmin'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:309: undefined reference to `atan'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:309: undefined reference to `sqrt'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:309: undefined reference to `sqrt'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:310: undefined reference to `sqrt'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:321: undefined reference to `sqrt'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `AngNormalize':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:142: undefined reference to `remainder'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `geod_genposition':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:529: undefined reference to `hypot'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:548: undefined reference to `atan2'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:548: undefined reference to `atan2'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:549: undefined reference to `atan2'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:549: undefined reference to `atan2'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:606: undefined reference to `atan2'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:474: undefined reference to `sincos'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:482: undefined reference to `sincos'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:511: undefined reference to `sincos'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:550: undefined reference to `atan2'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `AngNormalize':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:142: undefined reference to `remainder'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:142: undefined reference to `remainder'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:142: undefined reference to `remainder'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `geod_genposition':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:520: undefined reference to `sqrt'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:511: undefined reference to `sqrt'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `AngNormalize':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:142: undefined reference to `remainder'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:142: undefined reference to `remainder'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:142: undefined reference to `remainder'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o: in function `transitdirect':
./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:1759: undefined reference to `remainder'
/usr/bin/ld: ./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:1759: undefined reference to `remainder'
/usr/bin/ld: CMakeFiles/geodsigntest.dir/geodsigntest.c.o:./obj-x86_64-linux-gnu/src/tests/./src/geodesic.c:1799: more undefined references to `remainder' follow
collect2: error: ld returned 1 exit status
make[3]: *** [src/tests/CMakeFiles/geodsigntest.dir/build.make:100: bin/geodsigntest] Error 1
make[3]: Leaving directory '/build/proj-9.1.0~rc1/obj-x86_64-linux-gnu'
make[2]: *** [CMakeFiles/Makefile2:1572: src/tests/CMakeFiles/geodsigntest.dir/all] Error 2
make[2]: *** Waiting for unfinished jobs....
```
[Full buildlog](https://github.com/OSGeo/PROJ/files/9433655/proj_9.1.0.rc1_amd64.txt)

#### Problem description

`-lm` seems to be missing for `sqrt()` at least.

Might be caused by changes in glibc 2.34.

**Comment by sebastic** [comment]
The `check_function_exists` & `check_library_exists` calls in `cmake/ProjConfig.cmake` all fail:
```
-- Looking for localeconv
-- Looking for localeconv - not found
-- Looking for strerror
-- Looking for strerror - not found
-- Looking for dladdr in dl
-- Looking for dladdr in dl - not found
-- Looking for exp in m
-- Looking for exp in m - not found
```

This patch works around the issue:
```diff
--- a/src/tests/CMakeLists.txt
+++ b/src/tests/CMakeLists.txt
@@ -6,9 +6,7 @@ target_link_libraries(geodtest ${PROJ_LI
 add_executable(geodsigntest geodsigntest.c)
 # geodsigntest includes geodesic.c directly so it doesn't need to link
 # against the library.
-if(HAVE_LIBM)
-  target_link_libraries(geodsigntest m)
-endif()
+target_link_libraries(geodsigntest m)
 
 # Do not install, instead run tests
 add_test(NAME geodesic-test COMMAND geodtest)
```

**Comment by rouault** [comment]
@sebastic I've tried to reproduce your issue with the debian:unstable Docker image as best as I could, but I can't. Here's the output I get from cmake:
```
#  cmake -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=None -DCMAKE_INSTALL_SYSCONFDIR=/etc -DCMAKE_INSTALL_LOCALSTATEDIR=/var -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DFETCHCONTENT_FULLY_DISCONNECTED=ON -DCMAKE_INSTALL_RUNSTATEDIR=/run "-GUnix Makefiles" -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_INSTALL_LIBDIR=lib/x86_64-linux-gnu -DRUN_NETWORK_DEPENDENT_TESTS=OFF ..
-- The C compiler identification is GNU 12.2.0
-- The CXX compiler identification is GNU 12.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Requiring C++11
-- Requiring C++11 - done
-- Requiring C99
-- Requiring C99 - done
-- Configuring PROJ:
-- PROJ_VERSION                   = 9.1.0
-- Looking for localeconv
-- Looking for localeconv - found
-- Looking for strerror
-- Looking for strerror - found
-- Looking for dladdr in dl
-- Looking for dladdr in dl - found
-- Looking for exp in m
-- Looking for exp in m - found
[...]
```
The libc6 package I use is at 2.34-6
What would be different in the build environment you use compared to the one I use ? Just patching for libm detection doesn't seem to be the right thing to do, as we should definitely find those 4 symbols on Linux systems and building without localeconv, strerror and dladdr will result in a non-nominal build.

**Comment by sebastic** [comment]
> What would be different in the build environment you use compared to the one I use ?

Buildflags for one:
```shell
$ DEB_BUILD_MAINT_OPTIONS=hardening=+all dpkg-buildflags --export=sh
export ASFLAGS=""
export CFLAGS="-g -O2 -ffile-prefix-map=/tmp/buildd/proj=. -fstack-protector-strong -Wformat -Werror=format-security"
export CPPFLAGS="-Wdate-time -D_FORTIFY_SOURCE=2"
export CXXFLAGS="-g -O2 -ffile-prefix-map=/tmp/buildd/proj=. -fstack-protector-strong -Wformat -Werror=format-security"
export DFLAGS="-frelease"
export FCFLAGS="-g -O2 -ffile-prefix-map=/tmp/buildd/proj=. -fstack-protector-strong"
export FFLAGS="-g -O2 -ffile-prefix-map=/tmp/buildd/proj=. -fstack-protector-strong"
export GCJFLAGS="-g -O2 -ffile-prefix-map=/tmp/buildd/proj=. -fstack-protector-strong"
export LDFLAGS="-Wl,-z,relro -Wl,-z,now"
export OBJCFLAGS="-g -O2 -ffile-prefix-map=/tmp/buildd/proj=. -fstack-protector-strong -Wformat -Werror=format-security"
export OBJCXXFLAGS="-g -O2 -ffile-prefix-map=/tmp/buildd/proj=. -fstack-protector-strong -Wformat -Werror=format-security"
```

There is also no network in the build chroots, a limited path, and non-existent homedirectory for the build user:
```shell
env PATH="/usr/sbin:/usr/bin:/sbin:/bin" HOME="/nonexistent" dpkg-buildpackage -us -uc
```

**Comment by sebastic** [comment]
The build environment setup used in the Debian GIS team is documented in the team policy:

 https://debian-gis-team.pages.debian.net/policy/packaging.html#git-pbuilder

The procedure to build the package in the chroot interactively is documented more extensively in a postgis issue:

 https://trac.osgeo.org/postgis/ticket/5014#comment:3

**Comment by rouault** [comment]
ok, so actually the trigger for the issue was -Werror=format-security in CFLAGS.
Fix in https://github.com/OSGeo/PROJ/pull/3302

**Comment by sebastic** [comment]
Confirmed fixed with the changes in #3302.
